### PR TITLE
chore: upgrade istio to 1.4.10-asm.15

### PR DESCRIPTION
**Pull request by Bobgy**
**Description of your changes:**
I think I recently received a GCP email mentioning there's a vulnerability in some istio version, and we should upgrade to latest.

**Checklist:**
- [x] Unit tests have been rebuilt: 
    1. `cd manifests/tests`
    2. `make generate-changed-only`
    3. `make test`

### CODE REVIEW

File: tests/testdata_gcp_kpt/gcp/Kptfile [hunk 1]
```diff
@@ -11,37 +11,31 @@ openAPI:
         setter:
           name: gcloud.core.project
           value: customerProject
-          isSet: true
```
Comment (Bobgy) [inline tests/testdata_gcp_kpt/gcp/Kptfile:14]: I ran the hack/generate-xxx script, but looks like this was automatically removed

Comment (jlewi) [reply tests/testdata_gcp_kpt/gcp/Kptfile:14]: Can you revert the changes? You shouldn't need to rerun the generate script; it was a one time deal.

Comment (Bobgy) [reply tests/testdata_gcp_kpt/gcp/Kptfile:14]: Does that mean I don't need to update the testdata for kpt anymore?

**Comment by Bobgy** [comment]
Thanks @jlewi!
reverted unneeded changes

**Comment by Bobgy** [comment]
@jlewi We missed this for the release.

**Comment by Bobgy** [comment]
Reading the instruction
> Please upgrade your clusters to ASM 1.4.10-asm.3 or ASM 1.5.7-asm.0 as soon as possible and apply the additional configuration changes as specified in [ISTIO-SECURITY-2020-007](https://istio.io/latest/news/security/istio-security-2020-007/) to prevent Denial of Service (DOS) attacks on your application.

I don't think this is a blocker to release, because DOS attacks should be protected by IAP already. 
I'd prefer updating to latest istio, because we are not really providing guidelines on how users can upgrade by themselves.

**Comment by jlewi** [comment]
/lgtm
/approve

@Bobgy Note at this point 1.1.0 has been finalized.

**Pull request merged**